metaflow/plugins/argo/generate_input_paths.py

```diff
@@ -0,0 +1,23 @@
+import sys
+from hashlib import md5
+
+
+def generate_input_paths(step_name, timestamp, input_paths, max_splits):
+    # => run_id/step/:foo,bar
+    run_id = input_paths.split("/")[0]
+    foreach_base_id = "{}-{}-{}".format(step_name, timestamp, input_paths)
+
+    ids = [_generate_task_id(foreach_base_id, i) for i in range(int(max_splits))]
+    return "{}/{}/:{}".format(run_id, step_name, ",".join(ids))
+
+
+def _generate_task_id(base, idx):
+    # For foreach splits generate the expected input-paths based on max_split and base_id via.
+    # newline required at the end due to 'echo' appending one in the shell side task_id creation.
+    task_str = "%s-%s\n" % (base, idx)
+    hash = md5(task_str.encode("utf-8")).hexdigest()[-8:]
```

Comment:
it might be desirable to get task ids of the nature - `t-<hash>-<idx>` - in that scenario - we can probably get a much tighter compression of input_paths (assuming we change the input_paths compression/decompression logic) in addition to the UI looking a lot more nicer.